db/sql_exec_with_router.go

```diff
@@ -0,0 +1,109 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type DataSourceRouter interface {
+	GetDataSource(ctx context.Context) (DBConnectionPool, error)
+	GetAllDataSources() ([]DBConnectionPool, error)
```

Comment:
Quick question, how would we know which data source belongs to which Tenant? Would it be better to return a `map`?

Reply:
Currently I don't need it to return a map because I'm using it for closing all connections regardless of the tenant. When that becomes needed, we can switch it / refactor it.